### ISSUE TITLE
no kind "OpenStackMachineTemplate" is registered

### ISSUE THREAD

**Issue by okozachenko1203** [kubernetes-sigs/cluster-api-provider-openstack#2345]
/kind bug

**What steps did you take and what happened:**
I faced this error in one of our production cluster when i tried to get the list of osmt.
```sh
Error from server: conversion webhook for infrastructure.cluster.x-k8s.io/v1alpha7, Kind=OpenStackMachineTemplate failed: no kind "OpenStackMachineTemplate" is registered for version "infrastructure.cluster.x-k8s.io/v1beta1" in scheme "pkg/runtime/scheme.go:100"
```

**What did you expect to happen:**
The list of osmt resources.

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

**Environment:**

- Cluster API Provider OpenStack version (Or `git rev-parse HEAD` if manually built): 0.9.0
- Cluster-API version: 1.6.0
- OpenStack version: 
- Minikube/KIND version: 
- Kubernetes version (use `kubectl version`): 1.28
- OS (e.g. from `/etc/os-release`): Ubuntu 22.04

**Comment by okozachenko1203** [comment]
i think it is already reported and has a fix?
https://github.com/kubernetes-sigs/cluster-api-provider-openstack/pull/2052

**Comment by okozachenko1203** [comment]
let me share my story for other users
The reason of my case was that we updated mcapi (a custom operator which manages the lifecycle of CAPO resources) without upgrading CAPO version properly. More specifically, the new mcapi version creates v1beta1 CRs of CAPO which is not supported by CAPO 0.9.0.
That is why i got this error
```sh
Error from server: conversion webhook for infrastructure.cluster.x-k8s.io/v1alpha7, Kind=OpenStackMachineTemplate failed: no kind "OpenStackMachineTemplate" is registered for version "infrastructure.cluster.x-k8s.io/v1beta1" in scheme "pkg/runtime/scheme.go:100"
```
Btw, the critical thing is this error was popping up like a bruteforce and it occupied kube-api server. As a result, garbage collector was failed to sync its cache in kube-controller-manager. And we started to have dangling completed pods(we have some very frequent cronjobs) and ended up as etcd cluster degrade.

After upgrade CAPO to 0.10.x, and the kube-api has been released and garbage collector started working.
